core/spec/models/spree/order_cancellations_spec.rb

```diff
@@ -1,6 +1,64 @@
 require 'spec_helper'
 
 describe Spree::OrderCancellations do
+  describe "#cancel_unit" do
+    subject { Spree::OrderCancellations.new(order).cancel_unit(inventory_unit) }
+    let(:order) { create(:shipped_order, line_items_count: 1) }
+    let(:inventory_unit) { order.inventory_units.first }
+
+    it "creates a UnitCancel record" do
+      expect { subject }.to change { Spree::UnitCancel.count }.by(1)
+
+      unit_cancel = Spree::UnitCancel.last
+      expect(unit_cancel.inventory_unit).to eq inventory_unit
+      expect(unit_cancel.reason).to eq Spree::UnitCancel::DEFAULT_REASON
+    end
+
+    it "cancels the inventory unit" do
+      expect { subject }.to change { inventory_unit.state }.to "canceled"
+    end
+
+    context "with a reason" do
+      subject { order.cancellations.cancel_unit(inventory_unit, reason: "some reason") }
+
+      it "sets the reason on the UnitCancel" do
+        expect { subject }.to change { Spree::UnitCancel.count }.by(1)
+        expect(Spree::UnitCancel.last.reason).to eq("some reason")
+      end
+    end
+
+    context "with a who" do
+      subject { order.cancellations.cancel_unit(inventory_unit, whodunnit: "some automated system") }
+
+      it "sets the user on the UnitCancel" do
+        expect { subject }.to change { Spree::UnitCancel.count }.by(1)
+        expect(Spree::UnitCancel.last.created_by).to eq("some automated system")
+      end
+    end
+  end
+
+  describe "#reimburse_units" do
+    subject { Spree::OrderCancellations.new(order).reimburse_units([inventory_unit]) }
+    let(:order) { create(:shipped_order, line_items_count: 1) }
+    let(:inventory_unit) { order.inventory_units.first }
+    let!(:default_refund_reason) { Spree::RefundReason.find_or_create_by!(name: Spree::RefundReason::RETURN_PROCESSING_REASON, mutable: false) }
+
+    it "creates and performs a reimbursement" do
+      expect { subject }.to change { Spree::Reimbursement.count }.by(1)
+
+      reimbursement = Spree::Reimbursement.last
+      expect(reimbursement.refunds.size).to eq 1
+    end
+
+    it "creates return items for the inventory units and accepts them" do
+      expect { subject }.to change { Spree::ReturnItem.count }.by(1)
+
+      return_item = Spree::ReturnItem.last
```

Comment:
I would prefer to get this from the return from `subject` rather than hitting the DB again.